src/shared/machine-image.c

```diff
@@ -483,9 +483,15 @@ int image_remove(Image *i) {
         switch (i->type) {
 
         case IMAGE_SUBVOLUME:
-                r = btrfs_subvol_remove(i->path, BTRFS_REMOVE_RECURSIVE|BTRFS_REMOVE_QUOTA);
-                if (r < 0)
-                        return r;
+
+                /* Let's unlink first, maybe it is a symlink? If that works we are happy. Otherwise, let's get out the
+                 * big guns */
+                if (unlink(i->path) < 0) {
```

Comment:
Hmm, it would be nicer to maybe fall back to btrfs_subvol_remove only for some errnos (`EISDIR`?) Otherwise it's possible to obscure EPERM and such. But I guess it's OK for this to go in as is.